integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/util/ApictlUtils.java

```diff
@@ -262,22 +276,27 @@ public static void deployAPI(String apiProjectName, String mgwEnv) throws MicroG
      * @param apiName name of the API (in api.yaml) to undeploy
      * @param apiVersion version of the API
      * @param mgwEnv name of the apictl mgw env the API was deployed
+     * @param vhost vhost of the API to be undeployed from
      * @throws MicroGWTestException if apictl was unable to undeploy the API
      */
-    public static void undeployAPI(String apiName, String apiVersion, String mgwEnv) throws MicroGWTestException {
+    public static void undeployAPI(String apiName, String apiVersion, String mgwEnv, String vhost) throws MicroGWTestException {
         String[] cmdArray = { MG, UNDEPLOY, API };
-        String[] argsArray = { NAME_FLAG, apiName, VERSION_FLAG, apiVersion, ENV_FLAG, mgwEnv };
```

Comment:
How about something like:
```
        String[] argsArray = { NAME_FLAG, apiName, VERSION_FLAG, apiVersion, ENV_FLAG, mgwEnv }
        if (StringUtils.isNotEmpty(vhost)) {
            argsArray = { NAME_FLAG, apiName, VERSION_FLAG, apiVersion, ENV_FLAG, mgwEnv, VHOST_FLAG, vhost};
        }
```

Reply:
Thought to handle other optional flags as well :)
ex: if we want to set gw env as optional - if it is empty, remove from all gateway enviroinments

Reply:
+1. The method concat(String[]... stringArrays) at the end of this file might become useful to create new arg arrays using the existing arrays, after checking whether args like vhost, gw envs exists.